python/app/dialog.py

```diff
@@ -1124,7 +1187,8 @@ def setup_entity_model_view(self, entity_data):
         entity_data["sort_proxy"].sort(0, QtCore.Qt.DescendingOrder)
 
         # set up model
-        entity_data["view"].setModel(entity_data["sort_proxy"])
+        if not entity_data["has_filter_menu"]:
```

Comment:
Again, instead of specifically checking for the has filter menu, we should just check if the view already has a model set or not. If it does not have a model, then set it.